### DISCUSSION TITLE
MigrationManager can't find migrations when running outside of module directory

### DISCUSSION THREAD

**Discussion by theelderbeever** [piccolo-orm/piccolo#471] [Q&A]
I tend to structure my projects in the following manner and run the application from `.` as shown in the tree below. This allows for relative imports inside the application and seemingly helps with name collisions. When attempting to run `piccolo migrations forwards <app_name>` from the top level a few errors crop up...

1. `piccolo` command won't find the conf. 
  - This can be addressed by setting `export PICCOLO_CONF="application.db.conf"
2. The migration manager can't find table models and can't find migration files
  - Respectively, these can be solved by adding `${PWD}` and `${PWD}/application/db/migrations` to `PYTHONPATH`.

These definitely aren't ideal solutions. Do you have any recommendations on better ways to accomplish solve this issue?

```console
.
├── Dockerfile
├── README.md
├── application
│   ├── __init__.py
│   ├── app.py
│   ├── db
│   │   ├── __init__.py
│   │   ├── app.py
│   │   ├── conf.py
│   │   ├── endpoints.py
│   │   ├── migrations
│   │   │   ├── 2022-03-17T18-36-17-337727.py
│   │   │   ├── 2022-03-17T18-36-29-112174.py
│   │   │   ├── README.md
│   │   │   ├── __init__.py
│   │   ├── models
│   │   │   └── __init__.py
│   │   └── templates
│   │       ├── base.html.jinja
│   │       └── home.html.jinja
│   ├── static
│   │   ├── favicon.ico
│   │   └── main.css
├── poetry.lock
├── pyproject.toml
└── tests

**Comment by dantownsend** [comment]
How about moving your `piccolo_conf.py` file to the root folder?  In your example, I think you've renamed this to `conf.py`.

Then the contents of the `conf.py` file would be something like:

```python
APP_REGISTRY = AppRegistry(
    apps=[
        "application.db.app",  # assuming app.py if your piccolo app file
    ]
)
```

**Reply by theelderbeever** [reply]
Alright so I moved The `conf.py` to my root directory and changed `PICCOLO_CONF=conf` and the registered app is `"application.db.app"`. This has solved a separate issue where the app has trouble finding the actual application. However, unfortunately the migration manager still fails with 
>The command failed.
No module named '2022-03-17T18-36-29-112174'

So it looks like the MigrationManager may still be having trouble finding the migrations.

**Reply by theelderbeever** [reply]
Ah figured it out.... I was setting my migrations directory using a `Path` object. I turned it into a `str` and it solved it. As a mild form of preference it would be nice to see `pathlib` be supported native in `piccolo`. It was introduced in `python 3.4` so it _should_ meet `piccolo`'s requirements.

```python
CURRENT_DIRECTORY = Path(__file__).parent

APP_CONFIG = AppConfig(
    app_name="icytools",
    migrations_folder_path=CURRENT_DIRECTORY / "migrations",
    table_classes=[
        BlockTable,
        LogTable,
        TransactionTable,
    ],
    migration_dependencies=[],
    commands=[],
)
```

**Reply by dantownsend** [reply]
Yeah, good idea - I'll change it so it can accept `pathlib.Path`.

**Reply by dantownsend** [reply]
https://github.com/piccolo-orm/piccolo/pull/474
